addons/mail/static/src/components/chat_window_manager/chat_window_manager_tests.js

```diff
@@ -1464,6 +1464,231 @@ QUnit.test('chat window: switch on TAB', async function (assert) {
+                is_pinned: true,
+                name: "channel3",
+                state: 'open',
+            }],
+        },
+    });
+    await this.start({
+        env: {
+            browser: {
+                innerWidth: 1920,
+            },
+        },
+    });
+    assert.containsN(
+        document.body,
+        '.o_ChatWindow .o_ComposerTextInput_textarea',
+        3,
+        "initialy, 3 chat windows should be present"
+    );
+    assert.containsNone(
+        document.body,
+        '.o_ChatWindow.o-folded',
+        "all 3 chat windows should be open"
+    );
+
+    const chatWindows = document.querySelectorAll('.o_ChatWindow');
+    await afterNextRender(() => {
+        chatWindows[2].querySelector('.o_ComposerTextInput_textarea').focus();
+    });
+    assert.strictEqual(
+        chatWindows[2].querySelector('.o_ComposerTextInput_textarea'),
+        document.activeElement,
+        "The leftmost chatWindow should have the focus"
+    );
+
+    await afterNextRender(() =>
+        triggerEvent(
+            chatWindows[2].querySelector('.o_ComposerTextInput_textarea'),
+            'keydown',
+            { key: 'Tab' },
+        )
+    );
+    assert.strictEqual(
+        chatWindows[1].querySelector('.o_ComposerTextInput_textarea'),
+        document.activeElement,
+        "after pressing tab on the chat window on its left, the chatWindow on the middle should have focus"
+    );
+
+    await afterNextRender(() =>
+        triggerEvent(
+            chatWindows[1].querySelector('.o_ComposerTextInput_textarea'),
+            'keydown',
+            { key: 'Tab' },
+        )
+    );
+    assert.strictEqual(
+        chatWindows[0].querySelector('.o_ComposerTextInput_textarea'),
+        document.activeElement,
+        "after pressing tab on the chat window on its left, the rightmost chatWindow should have focus"
+    );
+
+    await afterNextRender(() =>
+        triggerEvent(
+            chatWindows[0].querySelector('.o_ComposerTextInput_textarea'),
+            'keydown',
+            { key: 'Tab' },
+        )
+    );
+    assert.strictEqual(
+        chatWindows[2].querySelector('.o_ComposerTextInput_textarea'),
+        document.activeElement,
+        "the leftmost chatWindow should have the focus after pressing tab on the rightmost chat window"
+    );
+});
+
+QUnit.test('chat window: cycle between more than 2 chat windows (RTL)', async function (assert) {
+    /**
+     * InnerWith computation uses following info:
+     * ([mocked] global window width: @see `mail/static/src/utils/test_utils.js:start()` method)
+     * (others: @see mail/static/src/models/chat_window_manager/chat_window_manager.js:visual)
+     *
+     * - chat window width: 325px
+     * - start/end/between gap width: 10px/10px/5px
+     * - hidden menu width: 200px
+     * - global width: 1920px
+     *
+     * Enough space for 2 visible chat windows:
+     *  10 + 325 + 5 + 325 + 10 + 325 + 5 = 1000 < 1920
```

Comment:
Still need to apply all the same previous comments as for the other test in this one.